.werft/util/certs.ts

```diff
@@ -25,49 +25,47 @@ export class InstallCertificateParams {
         return san.length <= 63;
     })) {
-        throw new Error(`there is no subdomain + '${params.domain}' shorter or equal to 63 characters, max. allowed length for CN. No HTTPS certs for you! Consider using a short branch name...`);
+        throw new Error(`there is no subdomain + '${domain}' shorter or equal to 63 characters, max. allowed length for CN. No HTTPS certs for you! Consider using a shorter branch name...`);
     }
+}
+
+function applyCertificateManifest(werft: Werft, pathToTemplate: string, namespace: string, certNamespace: string, domain: string, subdomains: string[], shellOpts: ExecOptions) {
     var cmd = `set -x \
-    && cd ${path.join(params.pathToTemplate)} \
+    && cd ${path.join(pathToTemplate)} \
     && cp cert-manager_certificate.tpl cert.yaml \
-    && yq w -i cert.yaml metadata.name '${params.namespace}' \
-    && yq w -i cert.yaml spec.secretName '${params.namespace}' \
-    && yq w -i cert.yaml metadata.namespace '${params.certNamespace}' \
-    ${subdomains.map(s => `&& yq w -i cert.yaml spec.dnsNames[+] '${s+params.domain}'`).join('  ')} \
+    && yq w -i cert.yaml metadata.name '${namespace}' \
+    && yq w -i cert.yaml spec.secretName '${namespace}' \
+    && yq w -i cert.yaml metadata.namespace '${certNamespace}' \
+    ${subdomains.map(s => `&& yq w -i cert.yaml spec.dnsNames[+] '${s + domain}'`).join('  ')} \
     && kubectl apply -f cert.yaml`;
 
-    werft.log("certificate", "Kubectl command for cert creation: " + cmd)
-    exec(cmd, { ...shellOpts, slice: 'certificate' });
-
-    werft.log('certificate', `waiting until certificate ${params.certNamespace}/${params.namespace} is ready...`)
-    let notReadyYet = true;
-    for (let i = 0; i < 90 && notReadyYet; i++) {
-        werft.log('certificate', `polling state of ${params.certNamespace}/${params.namespace}...`)
-        const result = exec(`kubectl -n ${params.certNamespace} get certificate ${params.namespace} -o jsonpath="{.status.conditions[?(@.type == 'Ready')].status}"`, { ...shellOpts, silent: true, dontCheckRc: true, async: false });
-        if (result != undefined && result.code === 0 && result.stdout === "True") {
-            notReadyYet = false;
-            break;
-        }
-
-        await sleep(5000);
-    }
+    werft.log(shellOpts.slice, "Kubectl command for cert creation: " + cmd)
+    exec(cmd, { ...shellOpts, slice: shellOpts.slice });
 }
 
-
 export async function installCertficate(werft, params: InstallCertificateParams, shellOpts: ExecOptions) {
     let notReadyYet = true;
-    werft.log('certificate', `copying certificate from "${params.certNamespace}/${params.certName}" to "${params.destinationNamespace}/${params.certSecretName}"`);
+    werft.log(shellOpts.slice, `copying certificate from "${params.certNamespace}/${params.certName}" to "${params.destinationNamespace}/${params.certSecretName}"`);
     const cmd = `kubectl get secret ${params.certName} --namespace=${params.certNamespace} -o yaml \
```

Comment:
I assume this uses the wrong kubectl context by default: it uses k3s but it should get the cert from core-dev.

Reply:
it depends on when you're calling this method...

The k3s kubeconfig is never downloaded if deploying without the `with-vm` annotation isn't passed, so the certificates are installed on the correct place